src/main/java/com/mabl/integration/jenkins/MablStepBuilder.java

```diff
@@ -214,6 +228,19 @@ private EnvVars getEnvironmentVars(Run<?, ?> build, TaskListener listener) {
         return environmentVars;
     }
 
+    static Secret getRestApiSecret(final String restApiKey) {
+        Secret secretKey = null;
+        List<UsernamePasswordCredentials> passwordCreds =
```

Comment:
Is there a reason we used username/password credential rather than secret text?
https://www.jenkins.io/doc/book/using/using-credentials/

Reply:
Yes. This one is available in Jenkins 1.x - secret text was not an option then.

Reply:
Are we still concerned about 1.x, looks like last release was 

1.651.3/ | 2016-06-14 06:20

Reply:
The problem is that we do not know for sure. With this change, we will now track the Jenkins versions that customers are using as well as the JVM version, too. One option for me is to go back and make 2.x the minimum for this Jenkins plugin and require Java 8. This changeset only moves us from Java 6 to Java 7 and sets the minimum from 1.580.1 to 1.625.1. What do you think? We just simply don't have the data about customers using the Jenkins plugin and know what version fo Jenkins they are using.

Reply:
I guess we can keep 1.x support until we figure out if anyone is using it, but I would prefer secret text to be supported as well so that people on 2.x do not have to specify username.

Reply:
I think allowing ~month to collect usage data is sensible, so we don't discover the hard way an important user is on a wood burning version of Java.

I do find it confusing that only user creds are allowed, but the UI doesn't intimate that. I initially tried making a "Secret Text" secret, only to notice it wouldn't show up in the mabl plugin step form.
![image](https://user-images.githubusercontent.com/1573793/83453090-dad08580-a427-11ea-8ed5-8c37479796a0.png)

Do we know what happens if you're a mabl user, on the current `0.0.19` version, that does not use "Credentials", but rather "configs". Will you automatically get a new Username/password credential created and populated in the mabl config for you, or will your jobs stop running until that's added?

If we must wait until we move to Jenkins 2.x, perhaps it make sense to merge the bug fixes, better telemetry, and other fixes here, but wait on changing the credential, so we can just move directly to the "Secret Text" type?

Reply:
Good point on the credential type. I can update the help text to be more explicit.

I tested an upgrade from a 0.0.19 installation. You need to go into editing the job and it will prompt you for an API key. If you select the same key as you used before, then all the settings are kept.

It seems like looking at the usage data - https://stats.jenkins.io/pluginversions/mabl-integration.html - that we can safely move to 2.x. So, I am gonna update this PR and require the 2.x as the minimum and use the Secret text credentials kind.